File: v3/typescript/api/authors-api.ts
import { AuthorsApiGenerated, AuthorsApiGetRequest, AuthorsApiPostRequest } from "./authors-api-generated";
import { searchAll } from "../search-all-helper";
import { AxiosRequestConfig } from "axios";
export * from "./authors-api-generated";

export class AuthorsApi extends AuthorsApiGenerated {

    public async get_all(requestParameters: AuthorsApiGetRequest, options?: AxiosRequestConfig) {
        return searchAll(this.get.bind(this), requestParameters, options);
    }

    public async post_all(requestParameters: AuthorsApiPostRequest, options?: AxiosRequestConfig) {
        return searchAll(this.post.bind(this), requestParameters, options);
    }
}

File: v3/python/newscatcherapi_client/apis/tags/authors_api.py
# coding: utf-8

from newscatcherapi_client.apis.tags.authors_api_generated import AuthorsApiGenerated
from newscatcherapi_client.search_all_helper import search_all

class AuthorsApi(AuthorsApiGenerated):

    def get_all(self, **kwargs):
        return search_all(self.get, **kwargs)

    def post_all(self, **kwargs):
        return search_all(self.post, **kwargs)
